plugins/com.google.cloud.tools.eclipse.appengine.validation/src/com/google/cloud/tools/eclipse/appengine/validation/ToServlet25QuickFix.java

```diff
@@ -39,14 +42,28 @@
   public String getLabel() {
     return Messages.getString("convert.servlet");
   }
-
+  
+  /**
+   * Attempts to edit the {@link IDocument} in the open editor. If the editor is not open,
+   * reads the file from memory and transforms in place.
+   */
   @Override
   public void run(IMarker marker) {
     try {
       IFile file = (IFile) marker.getResource();
-      URL xslPath = ToServlet25QuickFix.class.getResource("/xslt/servlet.xsl");
-      Xslt.transformInPlace(file, xslPath);
-    } catch (IOException | CoreException| TransformerException ex) {
+      IDocument document = ValidationUtils.getCurrentDocument(file);
+      URL xslPath = ApplicationQuickFix.class.getResource("/xslt/servlet.xsl");
+      if (document != null) {
+        String currentContents = document.get();
```

Comment:
We should definitely combine some of this duplicate code into one place or another. Maybe a superclass XsltQuickfix since the only difference is the stylesheet applied?